servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/NettyPipelineSslUtils.java

```diff
@@ -45,11 +52,55 @@ private NettyPipelineSslUtils() {
      *
      * @param pipeline The pipeline to check.
      * @return {@code true} if the pipeline is configured to use SSL/TLS.
+     *
+     * @deprecated not required anymore, will be removed in the future releases, see
+     * {@link #extractSslSessionAndReport(SslConfig, ChannelPipeline, ConnectionObserver)} for an alternative approach
      */
-    public static boolean isSslEnabled(ChannelPipeline pipeline) {
+    @Deprecated
+    public static boolean isSslEnabled(ChannelPipeline pipeline) {  // FIXME: 0.43 - remove deprecated method
         return pipeline.get(SslHandler.class) != null || pipeline.get(SniHandler.class) != null;
     }
 
+    /**
+     * Extracts the {@link SSLSession} from the {@link ChannelPipeline} if the handshake is already done
+     * and reports the result to {@link SecurityHandshakeObserver} if available.
+     *
+     * @param sslConfig {@link SslConfig} if SSL/TLS is expected
+     * @param pipeline {@link ChannelPipeline} which contains a handler containing the {@link SSLSession}
+     * @param connectionObserver {@link ConnectionObserver} in case the handshake status should be reported
+     * @return The {@link SSLSession} or {@code null} if none can be found
```

Comment:
consider adding @throws for IllegalStateException if the handler could not be found